crypto/ec/ec_key.c

```diff
@@ -368,12 +433,10 @@ int EC_KEY_set_public_key_affine_coordinates(EC_KEY *key, BIGNUM *x,
         goto err;
 
     /*
-     * Check if retrieved coordinates match originals and are less than field
-     * order: if not values are out of range.
+     * Check if retrieved coordinates match originals. The range check is done
+     * inside EC_KEY_check_key().
```

Comment:
EC_KEY_check_key just seems to call ec_key_simple_check_key(), so this function.

Reply:
I dont understand your comment.
EC_KEY_check_key(key)  is called just a few lines down from the comment, (so the range check was moved into the check function) . The test should be done in the check function since that is supposed to be tested as part of key validation..

Reply:
This is clearly EC_KEY_set_public_key_affine_coordinates(), and not ec_key_simple_check_key(), where all your other changes where.